### PR TITLE
Switch to kotlin gradle dsl and use version catalogs

### PR DESCRIPTION
**Pull request by viztea**
* switches from groovy to kotlin for most gradle files
* uses version catalogs
* updates gradle version to latest

### CODE REVIEW

**Comment by viztea** [comment]
~~I changed some annotation imports due to them not being found anymore, may have been an issue with my IDE, not sure. I'll see if I can revert it and still be able to compile~~

**Comment by melike2d** [comment]
fuck

File: LavalinkServer/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+tasks {
+    build {
+        doLast {
+            println("Version: $version")
+        }
+    }
+
+    processResources {
+        val buildNumber = if (System.getenv("CI") != null) {
```
Comment (devoxin) [inline LavalinkServer/build.gradle.kts:74]: Could this not be changed to something like
```kt
System.getenv("CI")?.let { System.getenv("BUILD_NUMBER") ?: "Unknown" } ?: "Unofficial"
```

Comment (freyacodes) [reply LavalinkServer/build.gradle.kts:74]: I think the original proposal is more readable

**Review by freyacodes — APPROVED** [review]
I was not aware of the version catalog feature. That's a welcome addition

**Pull request merged**